### PR TITLE
Log version on startup

### PR DESCRIPTION
**Pull request by karottenreibe**
Addresses issue [TS-36668](https://cqse.atlassian.net/browse/TS-36668)

- [x] Changes are tested adequately
- [x] Agent's README.md updated in case of user-visible changes
- [x] CHANGELOG.md updated
- [x] Present new features in [N&N](https://cqse.atlassian.net/l/cp/KHSr81m1)
- [x] [TGA Tutorial](https://docs.teamscale.com/tutorial/setting-up-tga-java) updated
- [x] [TIA Tutorial](https://docs.teamscale.com/tutorial/tia-java/) updated

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: system-tests/log-version-on-startup-test/src/main/java/jul/test/SystemUnderTest.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * The {@code SystemUnderTest} class is responsible for asserting that the LogManager is not initialized yet.
+ */
+public class SystemUnderTest {
+
+	public static boolean logManagerInitialized = false;
+
+	public static void main(String[] args) {
+		if (logManagerInitialized) {
+			throw new RuntimeException("Expected the LogManager to not be initialized by the agent. " +
+					"This is required to make the agent work in certain contexts like JBoss Wildfly. " +
```
Comment (DreierF) [inline system-tests/log-version-on-startup-test/src/main/java/jul/test/SystemUnderTest.java:13]: This is not really needed for the test scenario

File: teamscale-maven-plugin/pom.xml [hunk 2]
```diff
@@ -54,7 +54,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <revision>1.0.0-SNAPSHOT</revision>
-        <teamscale.agent.version>32.4.2</teamscale.agent.version>
+        <teamscale.agent.version>29.0.0</teamscale.agent.version>
```
Comment (DreierF) [inline teamscale-maven-plugin/pom.xml:57]: Does this need to be 29.0.0? If so I assume the change might have originated from renovate and we should disable it for this file and this dependency to no longer cause an automated update of it

**Review by DreierF — CHANGES_REQUESTED** [review]

File: teamscale-maven-plugin/pom.xml (hunk 2 above)
Comment (karottenreibe) [reply teamscale-maven-plugin/pom.xml:57]: I don't think that's renovate. Something in the build process is overwriting this value in this file whenever I run tests locally. Haven't figured out yet what that is. Sorry for accidentally checking this in

Comment (DreierF) [reply teamscale-maven-plugin/pom.xml:57]: 💡 Ah this is https://github.com/cqse/teamscale-jacoco-agent/blob/master/teamscale-maven-plugin/build.gradle.kts#L12

Comment (DreierF) [reply teamscale-maven-plugin/pom.xml]: Maybe a comment helps to avoid this confusion

File: teamscale-maven-plugin/pom.xml [hunk 3]
```diff
@@ -54,7 +54,7 @@
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
         <revision>1.0.0-SNAPSHOT</revision>
-        <teamscale.agent.version>29.0.0</teamscale.agent.version>
+        <teamscale.agent.version>32.4.2</teamscale.agent.version>
```
Comment (DreierF) [inline teamscale-maven-plugin/pom.xml:57]: Missed to commit the revert of this change?

**Pull request merged**